yarn-project/simulator/src/client/private_execution.ts

```diff
@@ -76,5 +72,7 @@ export async function executePrivateFunction(
     enqueuedPublicFunctionCalls,
     encryptedLogs,
     unencryptedLogs,
+    allEncryptedLogs: context.getAllEncryptedLogs(),
+    allUnencryptedLogs: context.getAllUnencryptedLogs(),
```

Comment:
Do we still need encryptedLogs/unencryptedLogs? 
Wonder if we should only assign all(Un)EncryptedLogs to the top level execution result 🤔

Reply:
Or emitting the counter (and note hash counter) along with the log preimage via oracle, then we can gather all logs from all nested calls to sort and squash them. I think you've suggested this? Sorry I am being slow. This is probably better than outputting them all in one go.

Reply:
> Do we still need encryptedLogs/unencryptedLogs?

I'm not sure really, but I'm keeping them as some tests rely on those logs being specific to the inidvidual call (i.e. not inlcuding extra logs from nested child/parent calls). Protocol-wise maybe it would be good to still keep track of logs per call to avoid confusion?

> Or emitting the counter (and note hash counter) along with the log preimage via oracle, then we can gather all logs from all nested calls to sort and squash them. 

Yep, I can do this! I just went for this method, because it seemed like the cleanest way of utilising the logs cache (which is intended mostly for removing transient logs) without adding too much extra code.

In this PR I can make the oracles also emit the log counter, then gather them from nested executions rather than populate a big array. The next PR will tackle separating note hash logs anyway, so I can add the note hash counter as part of that?

Reply:
Have implemented the above ^